quarkus-workshop-super-heroes/docs/src/docs/asciidoc/5-rest-client/rest-clients.adoc

```diff
@@ -236,103 +238,92 @@ To avoid this, we need to Mock the `HeroProxy` and `VillainProxy` interfaces.
 Alternatively, a convenient `io.quarkus.test.Mock` stereotype annotation could be used.
 This built-in stereotype declares `@Alternative`, `@Priority(1)` and `@Dependent`.
-So, to mock the `HeroProxy` interface we just need to implement the following `MockHeroProxy` class (under the `client` subpackage):
+So, to mock the `VillainProxy` interface we just need to implement the following `MockVillainProxy` class (under the `client` subpackage):
+
 
 [source,java]
 ----
-package io.quarkus.workshop.superheroes.fight.client;
+include::../../../../../super-heroes/rest-fights/src/test/java/io/quarkus/workshop/superheroes/fight/client/MockVillainProxy.java[]
+----
 
-import io.quarkus.test.Mock;
-import org.eclipse.microprofile.rest.client.inject.RestClient;
-
-import javax.enterprise.context.ApplicationScoped;
-
-@Mock
-@ApplicationScoped
-@RestClient
-public class MockHeroProxy implements HeroProxy {
-
-    public static final String DEFAULT_HERO_NAME = "Super Baguette";
-    public static final String DEFAULT_HERO_PICTURE = "super_baguette.png";
-    public static final String DEFAULT_HERO_POWERS = "eats baguette really quickly";
-    public static final int DEFAULT_HERO_LEVEL = 42;
-
-    @Override
-    public Hero findRandomHero() {
-        Hero hero = new Hero();
-        hero.name = DEFAULT_HERO_NAME;
-        hero.picture = DEFAULT_HERO_PICTURE;
-        hero.powers = DEFAULT_HERO_POWERS;
-        hero.level = DEFAULT_HERO_LEVEL;
-        return hero;
-    }
-}
+We are using some common classes for the test data.
+Let your IDE create the classes for you, and then fill in the following contents:
+
+[source,java]
+----
+include::../../../../../super-heroes/rest-fights/src/test/java/io/quarkus/workshop/superheroes/fight/client/DefaultTestHero.java[]
+----
+
+[source,java]
 ----
+include::../../../../../super-heroes/rest-fights/src/test/java/io/quarkus/workshop/superheroes/fight/client/DefaultTestVillain.java[]
+----
+
+We could do the same for the `HeroProxy`, but there are some drawbacks to using `@Mock`. Any class defined by `@Mock`
+is global in scope, and can't be isolated to individual tests. That can result in unwanted cross-talk between tests.
+Because of how the mock is defined,
+it's also harder to use https://site.mockito.org/[Mockito] to generate the mock instance.
+
+Let's try a different approach.
 
-icon:hand-point-right[role="red", size=2x] [red big]#Call to action#
+Add the extended Quarkus Mockito support to the `fight` service's `pom.xml`:
+
+[source,xml]
+----
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-junit5-mockito</artifactId>
+            <scope>test</scope>
+        </dependency>
+----
 
-Do the same for the `MockVillainProxy`:
+Now we can add the following to the top of the `FightResourceTest`:
 
 [source,java]
 ----
-package io.quarkus.workshop.superheroes.fight.client;
+    @InjectMock(convertScopes = true)
```

Comment:
It's only in the codebase because of lazy copying and pasting ... *but* I checked, and it's needed to mock singleton beans. And this line in the config means HeroProxy actually is a singleton: 

```
io.quarkus.workshop.superheroes.fight.client.HeroProxy/mp-rest/scope=javax.inject.Singleton
```

So it looks like I got lucky!

Reply:
So then my follow-up question - why is it a singleton and not default application scoped?

Reply:
No idea. :) 

But I think it's outside the scope of this module ...